### PR TITLE
Booking controller update and test added

### PR DESCRIPTION
**Pull request by adrianchen91**
Updating booking creation process to include test. 
Controller test added. - Only working tests included, non functioning tests removed in this PR

bootstrap table package added in preparation to change booking timetable look

### CODE REVIEW

File: server/controllers/booking/booking.controller.js [hunk 1]
```diff
@@ -43,13 +50,60 @@ const createBooking = function(req, res) {
...
-                res.status(504).send(err);
-              } else {
-                res.json(booking._id);
-              }
-            });
+            Booking.find({ car: req.body.car, disabled: false })
+              // Placeholder, should look for booking belonging to current user
+              .exec((err, bookings) => {
+                if (err) {
+                  res.status(500).send(err);
```
Comment (Rukia3d) [inline server/controllers/booking/booking.controller.js]: Looks like it's all OK you do not create a new booking. Is it intentional?

Comment (adrianchen91) [reply server/controllers/booking/booking.controller.js]: No :) Great catch. I should definitely create a booking if no booking is found. I will fix that later tonight

File: server/controllers/booking/booking.controller.js [hunk 2]
```diff
@@ -43,13 +50,66 @@ const createBooking = function(req, res) {
...
             res.status(500).send(errs);
           } else {
-            newBooking.save((err, booking) => {
-              if (err) {
-                res.status(504).send(err);
-              } else {
-                res.json(booking._id);
-              }
-            });
+            Booking.find({ car: req.body.car, disabled: false })
```
Comment (Rukia3d) [inline server/controllers/booking/booking.controller.js]: You don't need to check bookings for current users - only bookings for this car. You can add times as parameters for the request like here https://github.com/carshareadm/carshare/blob/master/server/controllers/cars/cars.controller.js#L60 and check if it returns null (no bookings) then you won't need that many checks

File: server/controllers/booking/booking.controller.js [hunk 3]
```diff
@@ -2,19 +2,26 @@ import Booking from "../../models/booking";
...
 
 const createBooking = function(req, res) {
   const startAt = req.body.startAt;
   const endAt = req.body.endAt;
   const carid = req.body.car;
   const userid = req.body.userid;
 
-  if(!userid || !carid)
-  {
+  if (
```
Comment (Rukia3d) [inline server/controllers/booking/booking.controller.js]: Both of these checks are the same, do you need 2 of them?

File: server/controllers/booking/booking.controller.js [hunk 4]
```diff
@@ -43,13 +50,66 @@ const createBooking = function(req, res) {
...
+                      if (err) {
+                        res.status(500).send(err);
+                      } else {
+                        res.status(200).send(booking._id);
+                      }
+                    });
+                  } else {
+                    //checking of time to be implemented
+                    bookings.forEach(b => {
+                      var compStart = moment(bookings.startsAt);
```
Comment (Rukia3d) [inline server/controllers/booking/booking.controller.js]: This won't work as you define b in forEach but use bookings to reference it

File: server/controllers/booking/booking.controller.js (hunk 2 above)
Comment (adrianchen91) [reply server/controllers/booking/booking.controller.js]: Updated

File: server/controllers/booking/booking.controller.js (hunk 3 above)
Comment (adrianchen91) [reply server/controllers/booking/booking.controller.js]: No, also corrected

File: server/controllers/booking/booking.controller.js (hunk 4 above)
Comment (adrianchen91) [reply server/controllers/booking/booking.controller.js]: thank you 😀

**Pull request merged**